### PR TITLE
Optimise fetch cache lookup

### PR DESCRIPTION
**Pull request by rwd**
Running the included benchmark script with `bundle exec ruby spec/benchmark.rb`:
* On the current master branch: `  0.880000   0.000000   0.880000 (  0.882051)`
* On this branch: `  0.220000   0.010000   0.230000 (  0.231535)`

### CODE REVIEW

File: lib/i18n_data.rb [hunk 1]
```diff
@@ -47,12 +47,13 @@ def data_provider=(provider)
...
       @cache ||= {}
-      if @cache.key?(args)
-        @cache[args]
+      @cache[type] ||= {}
+      if @cache[type].key?(language_code)
+        @cache[type][language_code]
       else
-        @cache[args] = yield
+        @cache[type][language_code] = yield
       end
```
Comment (grosser) [inline lib/i18n_data.rb]: how about: 
 - initialize `@cache` as `Hash.new { |h, k| h[k] = {} }`
 - `@cache[type].fetch(language_code) { @cache[language_code] = yield }`

Comment (rwd) [reply lib/i18n_data.rb]: That is more concise, and performant, so: yes!
`  0.180000   0.000000   0.180000 (  0.181957)`

Comment (rwd) [reply lib/i18n_data.rb]: (Added `[type]` to the contents of the `fetch` block, BTW.)

File: lib/i18n_data.rb [hunk 2]
```diff
@@ -48,13 +48,8 @@ def data_provider=(provider)
...
 
     def fetch(type, language_code)
-      @cache ||= {}
-      @cache[type] ||= {}
-      if @cache[type].key?(language_code)
-        @cache[type][language_code]
-      else
-        @cache[type][language_code] = yield
-      end
+      @cache ||= Hash.new { |h, k| h[k] = {} }
```
Comment (grosser) [inline lib/i18n_data.rb]: can this live in `def initialize` ?

File: lib/i18n_data.rb [hunk 3]
```diff
@@ -47,13 +47,9 @@ def data_provider=(provider)
...
-    def fetch(*args)
-      @cache ||= {}
-      if @cache.key?(args)
-        @cache[args]
-      else
-        @cache[args] = yield
-      end
+    def fetch(type, language_code)
+      @cache ||= Hash.new { |h, k| h[k] = {} }
+      @cache[type].fetch(language_code) { @cache[type][language_code] = yield }
```
Comment (grosser) [inline lib/i18n_data.rb]: might be faster to do 
```ruby
type_cache = @cache[type]
type_cache.fetch(language_code) { type_cache[language_code] = yield }
```

**Comment by grosser** [comment]
0.9.1 🎉

**Pull request merged**